File: LC_Gen_Codes/2843_no_setting.py
Here is the code implementation for the given problem:


class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        count = 0
        
        for i in range(low, high+1):
            num_str = str(i)
            
            # Numbers with an odd number of digits are never symmetric
            if len(num_str) % 2 != 0:
                continue
            
            n = len(num_str) // 2
            
            # Check if the sum of the first n digits is equal to the sum of the last n digits
            if sum(map(int, num_str[:n])) == sum(map(int, num_str[n:])):
                count += 1
        
        return count


The function `countSymmetricIntegers` takes in two parameters, `low` and `high`, representing the range of numbers to check for symmetric integers. It initializes a variable `count` to keep track of the number of symmetric integers.

The function then iterates over the range of numbers from `low` to `high+1`. For each number, it converts it to a string `num_str`.

If the length of `num_str` is odd, indicating that the number has an odd number of digits, it skips to the next iteration using the `continue` statement.

If the length of `num_str` is even, it calculates the midpoint `n` by dividing the length of `num_str` by 2.

It then checks if the sum of the first `n` digits, obtained by slicing `num_str` from index 0 to `n`, is equal to the sum of the last `n` digits, obtained by slicing `num_str` from index `n` to the end.

If the sum of the first `n` digits is equal to the sum of the last `n` digits, it increments the `count` variable by 1.

Finally, the function returns the `count`, which represents the number of symmetric integers in the given range.

File: LC_Gen_Codes_Java/2843_no_setting.java
Here is the converted code to Java:

java
class Solution {
    public int countSymmetricIntegers(int low, int high) {
        int count = 0;

        for (int i = low; i <= high; i++) {
            String numStr = String.valueOf(i);

            // Numbers with an odd number of digits are never symmetric
            if (numStr.length() % 2 != 0) {
                continue;
            }

            int n = numStr.length() / 2;

            // Check if the sum of the first n digits is equal to the sum of the last n digits
            int sumFirstHalf = 0;
            int sumSecondHalf = 0;
            for (int j = 0; j < n; j++) {
                sumFirstHalf += Integer.parseInt(String.valueOf(numStr.charAt(j)));
                sumSecondHalf += Integer.parseInt(String.valueOf(numStr.charAt(j + n)));
            }
            
            if (sumFirstHalf == sumSecondHalf) {
                count++;
            }
        }

        return count;
    }
}